webui/react/src/components/LogViewerTimestamp.tsx

```diff
@@ -0,0 +1,656 @@
+import { Button, notification, Space, Tooltip } from 'antd';
+import dayjs, { Dayjs } from 'dayjs';
+import React, {
+  useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState,
+} from 'react';
+import screenfull from 'screenfull';
+import { sprintf } from 'sprintf-js';
+import { throttle } from 'throttle-debounce';
+
+import Icon from 'components/Icon';
+import usePrevious from 'hooks/usePrevious';
+import useResize, { DEFAULT_RESIZE_THROTTLE_TIME } from 'hooks/useResize';
+import useScroll, { defaultScrollInfo } from 'hooks/useScroll';
+import { LogLevel, TrialLog } from 'types';
+import { formatDatetime } from 'utils/date';
+import { ansiToHtml, copyToClipboard, toRem } from 'utils/dom';
+import { capitalize } from 'utils/string';
+
+import { FetchArgs } from '../../../api-ts-sdk';
+import { TrialLogFiltersInterface } from '../pages/TrialLogs/TrialLogFilters';
+import { consumeStream } from '../services/utils';
+
+import css from './LogViewer.module.scss';
+import Page, { Props as PageProps } from './Page';
+
+export interface LogViewerTimestampFilterComponentProp {
+  filter: TrialLogFiltersInterface;
+  filterOptions: TrialLogFiltersInterface;
+  onChange?: (newFilters: TrialLogFiltersInterface) => void;
+}
+
+export interface LogViewerTimestampFilter {
+  limit?: number,
+  timestampAfter?: Dayjs,
+  timestampBefore?: Dayjs,
+}
+
+interface Props {
+  FilterComponent?: React.ComponentType<LogViewerTimestampFilterComponentProp>,
+  debugMode?: boolean;
+  disableLevel?: boolean;
+  fetchToLogConverter: (data: unknown) => TrialLog,
+  noWrap?: boolean;
+  onDownloadClick?: () => void;
+  onFetchLogAfter: (filters: LogViewerTimestampFilter, canceler: AbortController) => FetchArgs;
+  onFetchLogBefore: (filters: LogViewerTimestampFilter, canceler: AbortController) => FetchArgs;
+  onFetchLogFilter: (canceler: AbortController) => FetchArgs;
+  onFetchLogTail: (filters: LogViewerTimestampFilter, canceler: AbortController) => FetchArgs;
+  pageProps: Partial<PageProps>;
+}
+
+interface ViewerLog extends TrialLog {
+  formattedTime: string;
+}
+
+interface LogConfig {
+  charHeight: number;
+  charWidth: number;
+  dateTimeWidth: number;
+  messageSizes: Record<string, MessageSize>;
+  messageWidth: number;
+  totalContentHeight: number;
+}
+
+interface MessageSize {
+  height: number;
+  top: number;
+}
+
+export const TAIL_SIZE = 50;
+
+// What factor to multiply against the displayable lines in the visible view.
+const BUFFER_FACTOR = 1;
+
+// Format the datetime to...
+const DATETIME_PREFIX = '[';
+const DATETIME_SUFFIX = ']';
+const DATETIME_FORMAT = `[${DATETIME_PREFIX}]YYYY-MM-DD, HH:mm:ss${DATETIME_SUFFIX}`;
```

Comment:
**nit:** Remove the comma `,`. This came up during last friday's demo when @stoksc was trying to copy and paste the timestamp into the date time filters, without the comma the copy and paste works beautifully.

Reply:
this is the related ticket: https://determinedai.atlassian.net/browse/DET-4794